units/pod_allocator.go

```diff
@@ -107,9 +107,37 @@ func (j *podAllocatorJob) Run(ctx context.Context) {
 			j.AddRetryableError(errors.Wrap(err, "marking unallocatable container task as system-failed"))
 		}
 
+		grip.Info(message.Fields{
+			"message": "refusing to allocate task because it has no more allocation attempts",
+			"task":    j.task.Id,
+			"project": j.pRef.Identifier,
+			"job":     j.ID(),
+		})
+
 		return
 	}
 	if !j.task.ShouldAllocateContainer() {
+		grip.Info(message.Fields{
+			"message": "refusing to allocate task because it is not in a valid state to be allocated",
+			"task":    j.task.Id,
+			"project": j.pRef.Identifier,
+			"job":     j.ID(),
+		})
+		return
+	}
+
+	canDispatch, reason := model.ProjectCanDispatchTask(j.pRef, j.task)
+	if !canDispatch {
+		if reason == "" {
+			reason = "of unknown reason"
+		}
+		grip.Info(message.Fields{
+			"message": "refusing to allocate task because it is not dispatchable",
```

Comment:
This implies something is amiss with the task itself. Can we change it to say that it can't be allocated due to the project ref's settings?

Reply:
Done.